topicList: Add an OfflineNotice component.

codebu5ter opened the pull request:
This adds an OfflineNotice component to the topicList to let the user know that they are offline, instead of just showing the LoadingIndicator spinning endlessly.

![screenshot_20180414-110127](https://user-images.githubusercontent.com/16636569/38764801-56ee500e-3fd3-11e8-82ae-8076c3a46ad3.png)

EDIT: Updated the screenshot.

codebu5ter commented:
The padding around the OfflineNotice is due to the padding defined in the TopicListScreen. Shall I remove that padding?

borisyankov commented:
This approach is not very scalable. We will need to add this to every component manually.

codebu5ter commented:
@borisyankov I've now reworked my commit a bit. I've now added a showOfflineNotice prop to the Screen component, so that any screen requiring to show the OfflineNotice can just pass the showOfflineNotice prop to do so. This will scale much better and will require minimal effort to implement.

jainkuniya commented on src/common/Screen.js:
```diff
@@ -36,6 +36,7 @@ type Props = {
   search?: boolean,
   title?: LocalizableText,
   searchBarOnChange?: (text: string) => void,
+  showOfflineNotice?: boolean,
```
(clean up: just add it above method)

codebu5ter replied:
Done. 👍

jainkuniya commented on src/common/Screen.js:
```diff
@@ -60,6 +61,7 @@ class Screen extends PureComponent<Props> {
       search,
       searchBarOnChange,
       title,
+      showOfflineNotice,
```
(clean up: add in alphabetically)

codebu5ter replied:
Changes done, as requested. 👍

borisyankov commented:
@codebu5ter This PR has stalled a bit.
When reviewing it with @gnprice last week we thought it might be good to merge.

After a more careful examination though I think otherwise.

First thing, the code now is clear, concise and correct. Thanks for making it so.

My recommendation not to implement the offline notice this way is because we use it only once in the `TopicListScreen` and it doesn't look like we will need it in another `Screen`. Thus, just including the `<OfflineNotice />` manually in the `TopicListScreen` just above the `<TopicList />` will be simplest.

We might not even want to do this because we are considering moving the topics to another place in the UI (but no final decision made yet)

borisyankov commented:
@gnprice I think we can close this PR now.

The alternative suggested by me, to include the `<OfflineNotice />` in `TopicListScreen` is a better and shorter solution, but an even better solution, one that we discussed and should do relatively soon, is to correctly indicate a failed loading of the topic data, which will correlate with no connection.

borisyankov commented:
I decided to add the `<OfflineNotice />` anyway and after testing, it seems Robert did add this correctly during the summer! :)

https://github.com/zulip/zulip-mobile/blob/master/src/common/Screen.js#L111

This can now be closed as implemented!

gnprice commented:
:smile: Great!
